docs/api_specs/05_informational_api_specs.md

```diff
@@ -0,0 +1,388 @@
+# Watcher - Informational API Service
+
+API specification of the Watcher's Informational Service
+
+
+## Account - Get Balance
+
+```shell
+http POST /account.get_balance address=b3256026863eb6ae5b06fa396ab09069784ea8ea
+```
+
+```elixir
+// TODO
+```
+
+```javascript
+// TODO
+```
+
+> The above command returns JSON document:
+
+```json
+{
+    "version": "1",
+    "success": true,
+    "data": [
+        {
+            "currency": "BFDF85743EF16CFB1F8D4DD1DFC74C51DC496434",
+            "amount": 20
+        },
+        {
+            "currency": "0000000000000000000000000000000000000000",
+            "amount": 1000000000
+        }
+    ]
+}
+```
+
+Returns the balance of each currency for the given account address
+
+### HTTP Request
+
+`POST /account.get_balance`
+
+### JSON Body
+
+Key | Type | Description
+--------- | ------- | -----------
+address | Hex encoded string | Address of the funds owner
+
+
+
+## Account - Get Utxos
```

Comment:
I think currently it falls under `security critical`. Otherwise the user doesn't know what to spend or exit in case of a byzantine condition.

Honestly, figuring out how to draw the line here evades for a long time :/.

Reply:
Oof, this is a tough one!
You're right that the user will need to get their utxos to be able to exit.
But tracking utxos is going to affect the performance of the Watcher...

Maybe it could be in both places - the Informational API (need a better name for this) can always track and store utxos in its DB, so they can be quickly retrieved by e.g. a Wallet.

And Watcher doesn't maintain a list of utxos anywhere, but can provide a account.get_utxos call that can dynamically look them up. It's computationally more expensive, but should only happen in unusual circumstances.

Reply:
I think `looking for utxo only in unusual cases`  is OK. Would it make sense to not serve this functionality in "Security" & "Info API" under the same name? It will do the same, but the first one should make it clear `don't call me too often`.

Any way we'd need to store owner address in OMG.DB, computing it from signature for every lookup will be too much.

Reply:
yeah - agree on the dual nature of this endpoint - same API, different implementation and performance. In the `security critical` world, the UTXO set will just be brute-force scanned (order-of-magnitude millions of entries max - as determined by exit-ability implications of the protocol).

Mind however 

>we'd need to store owner address in OMG.DB

it is stored - every UTXO's value is `{amount, currency, owner}`, `owner` being the address.
When you have the UTXO positions necessary, you can grab the blocks, txbytes and calculate proof from that.